IO::Socket::IP: detect if setsockopt() fails to clear IPV6_V6ONLY

tonycoz opened the pull request:
dragonfly bsd 6.4 succeeds in clearing this, even though it doesn't clear it.

https://rt.cpan.org/Ticket/Display.html?id=148293

hvds commented:
The code looks entirely reasonable to me, but in the commit message I suggest `s{succeeds in clearing this}{claims to clear this}` or similar.

tonycoz committed:
IO::Socket::IP: detect if setsockopt() fails to clear IPV6_V6ONLY

dragonfly bsd 6.4 claims to clear this, even though it doesn't
clear it.

This is documented as a deliberate change in 6.4:

https://www.dragonflybsd.org/release64/

Reported upstream at:

https://rt.cpan.org/Ticket/Display.html?id=148293

tonycoz commented:
The new push rebased on blead, adjusts the commit wording and adds a reference to the dfly 6.4 release notes.
